packaging/installer/kickstart-ng.sh

```diff
@@ -0,0 +1,914 @@
+      fatal "Unsupported system type detected. Netdata cannot be installed on this system using this script."
+      ;;
+  esac
+}
+
+str_in_list() {
+  printf "%s\n" "${2}" | tr ' ' "\n" | grep -qE "^${1}\$"
+  return $?
+}
+
+confirm_root_support() {
+  if [ "$(id -u)" -ne "0" ]; then
+    if [ -z "${ROOTCMD}" ] && command -v sudo > /dev/null; then
+      ROOTCMD="sudo"
+    fi
+
+    if [ -z "${ROOTCMD}" ] && command -v doas > /dev/null; then
+      ROOTCMD="doas"
+    fi
+
+    if [ -z "${ROOTCMD}" ] && command -v pkexec > /dev/null; then
+      ROOTCMD="pkexec"
+    fi
+
+    if [ -z "${ROOTCMD}" ]; then
+      fatal "We need root privileges to continue, but cannot find a way to gain them. Either re-run this script as root, or set \$ROOTCMD to a command that can be used to gain root privileges"
+    fi
+  fi
+}
+
+# ======================================================================
+# Existing install handling code
+
+update() {
+  updater="${ndprefix}/usr/libexec/netdata/netdata-updater.sh"
+
+  if [ -x "${updater}" ]; then
+    if run ${ROOTCMD} "${updater}" --not-running-from-cron; then
+      progress "Updated existing install at ${ndprefix}"
+      return 0
+    else
+      fatal "Failed to update existing Netdata install at ${ndprefix}"
+    fi
+  else
+    return 1
+  fi
+}
+
+handle_existing_install() {
+  if pkg_installed netdata; then
+    ndprefix="/"
+  else
+    ndpath="$(command -v netdata 2>/dev/null)"
+    if [ -z "$ndpath" ] && [ -x /opt/netdata/bin/netdata ]; then
+      ndpath="/opt/netdata/bin/netdata"
+    fi
+
+    if [ -n "${ndpath}" ]; then
+      ndprefix="$(dirname "$(dirname "${ndpath}")")"
+    fi
+
+    if [ "${ndprefix}" = /usr ]; then
+      ndprefix="/"
+    fi
+  fi
+
+  if [ -n "${ndprefix}" ]; then
+    typefile="${ndprefix}/etc/netdata/.install-type"
+    if [ -r "${typefile}" ]; then
+      ${ROOTCMD} sh -c "cat \"${typefile}\" > \"${tmpdir}/install-type\""
+      # shellcheck disable=SC1091
+      . "${tmpdir}/install-type"
+    else
+      INSTALL_TYPE="unknown"
+    fi
+  fi
+
+  if [ -z "${ndprefix}" ]; then
+    progress "No existing installations of netdata found, assuming this is a fresh install."
+    return 0
+  fi
+
+  case "${INSTALL_TYPE}" in
+    kickstart-*|legacy-*|manual-static)
+      if [ -n "${NETDATA_REINSTALL}" ]; then
+        progress "Found an existing netdata install at ${ndprefix}, but user requested reinstall, continuing."
+        return 0
+      fi
+
+      ret=0
+
+      if [ "${NETDATA_CLAIM_ONLY}" -eq 0 ]; then
+        if ! update; then
+          warning "Unable to find usable updater script, not updating existing install at ${ndprefix}."
+        fi
+      fi
+
+      if [ -n "${NETDATA_CLAIM_TOKEN}" ]; then
+        progress "Attempting to claim existing install at ${ndprefix}."
+        claim
+        ret=$?
+      elif [ "${NETDATA_CLAIM_ONLY}" -eq 1 ]; then
+        fatal "User asked to claim, but did not proide a claiming token."
+      fi
+
+      exit $ret
```

Comment:
Should we add a message here? From a user point of view it can be:

```cmd
--- Using ${tmpdir} as a temporary directory. ---
# and the scripts exit here w/o any additional message
```

Reply:
Fixes for this are planned as part of the next PR (I’ve technically already got a fix for it, but it’s rather entangled with the changes in the same area that ensure that reinstalls preserve the install type, so it’s nontrivial to backport to this PR).

Reply:
Ok, decided to ask because it was quite confusing to me.